# eSports Over the Years - An Analysis of Video Game "Success" in eSports by Genre
by Abbas Syed
<br>
Student #: 100973637

---

## Introduction
A "sport" has traditionally been defined as any competitive-based activity involving physical, full-body movement. However, since the invention of the computer and later, video games, the term has slowly been 'widening' to accommodate other forms of competitive play. Gradually, as more competitive-style video games released (such as shooters, battle arenas, etc.), more people around the world started to realize their skills in these games and like-minded players begun hosting tournaments together. Communities formed, tournaments got bigger, groups and organizations dedicated to these tournaments were founded; this all accumulated to the eSports industry that it is today.
<br>
<br>
Despite the rapid growth it has seen since the beginning, the eSports industry is still relatively very young, where there still does not exist proper standardized regulations, administration, or a sole governing body. National governments also have varied involvement in supervising eSports tournaments and the like; in general, it is very inconsistent. This has caused concerns over the integrity of the industry and its future, with people questioning the credibility of finding a career or investing in eSports, and even whether or not any part of the eSports market is profitable. Numerous forms of research/media have analyzed the industry and discussed what eSports needs to be sustainable, and in this report, the question that is looking to be answered is: **"how does the genre of video game correlate to its success (or lack thereof) in eSports?"**
<br>
<br>
Video game genre is fundamental to the eSport treatment a game may receive in its lifetime. Recall that sports are by definition competitive—this requires the video game as well to at least include some sort of competitive aspect, in which players can compete for things like a higher score, faster elapsed time, more wins, etc. The genre of video game defines what kind of gameplay style is to be expected, and thus tells us whether or not the game is naturally competitive.
<br>
Common, generally non-competitive video game genres include:

- Adventure
- Interactive/Visual Novel
- Simulators
- Sandbox
- Open-world Exploration
- Horror/Survival
- Platformer

This is not an exhaustive list and each general genre contains its own specific sub-genres, however, it is not relevant to this report. Instead, we are concerned with the competitively-driven genres of video games. Based on the dataset used, the following list shows all of the video game genres that have seen activity in eSports:

- Strategy
- First-Person Shooter (FPS)
- Multiplayer Online Battle Arena (MOBA)
- Role-Playing Game (RPG)
- Fighting Game
- Racing
- Sports
- Collectible Card Game (sometimes referred to as "Trading Card Game" or "TCG")
- Puzzle Game
- Battle Royale
- Third-Person Shooter
- Music/Rhythm Game

The dataset is taken from [Kaggle.com](https://www.kaggle.com/datasets/rankirsh/esports-earnings/data), and is authored by Ran Kirshenboim, who mentions that the dataset sources from [EsportEarnings.com](https://www.esportsearnings.com/). Along with the video game name, it contains each game's release year, genre, total earnings (including offline and % of earnings offline), total players, and total number of tournaments (all within 1998 to 2023).

## Analyses
The CSV dataset will be analyzed using Pandas, Numpy, Matplotlib's PyPlot module and Seaborn. The packages have been installed into the project folder using Pip, within a virtual environment so as to remain local to the project. iPyKernel was also installed to use as the local Python kernel for Jupyter. First, let's import the mentioned packages as well as our CSV data.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sb

df = pd.read_csv("GeneralEsportData.csv")
df.head()

Game  ReleaseDate     Genre  TotalEarnings  \
0         Age of Empires         1997  Strategy      736284.75   
1      Age of Empires II         1999  Strategy     3898508.73   
2     Age of Empires III         2005  Strategy      122256.72   
3      Age of Empires IV         2021  Strategy     1190813.44   
4  Age of Empires Online         2011  Strategy       11462.98   

   OfflineEarnings  PercentOffline  TotalPlayers  TotalTournaments  
0        522378.17        0.709478           624               341  
1       1361409.22        0.349213          2256              1939  
2         44472.60        0.363764           172               179  
3        439117.93        0.368755           643               423  
4           775.00        0.067609            52                68

Here is a preview of the first 5 entries in the CSV table. The "Game", "ReleaseDate", and "Genre" are self-explanatory/have already been explained. Notice how different releases or iterations of games are treated as their own separate game. Sometimes this can be due to new features, implementations, fixes and more that the developers have added to the game (e.g. Blizzard Entertainment's Overwatch and Overwatch 2), other times it is because the next installation focuses on a different aspect from the previous game entirely, while maintaining the same gameplay features.
<br>
<br>
"TotalEarnings" depicts the total amount of earnings, in USD, that have been given out to players and teams for its respective game.
<br>
"OfflineEarnings" depicts the amount of earnings that were given out in "offline" tournaments, meaning only tournaments that were held in-person (sometimes called LAN tournaments, where LAN is short for local area network) and not online via servers.
<br>
"PercentOffline" depicts the percent as a decimal value of OfflineEarnings out of TotalEarnings, rounded to 6 decimal places.
<br>
"TotalPlayers" depicts not the entire playerbase but the total amount of players that participated in tournaments for the respective game.
<br>
"TotalTournaments" depicts the total amount of tournaments that were held for the respective game, both offline and online.
<br>
<br>
Next, some pre-processing and cleaning will be performed using Pandas to ready the data for proper analysis.

In [2]:
print(df.shape)

(669, 8)


`df.shape` returns a simple tuple that shows the number of entries (rows) in the table, and the number of value categories (columns). In our case, the categories will always remain constant at 8. Keep in mind that the data frame already considers the top row to be labels and only returns the number of rows with actual data. So, the original dataset has 669 entries.

In [3]:
df.drop_duplicates() # Removing any unneseccary duplicate entries
print(df[df["TotalTournaments"] == 0].shape) # Prints a tuple in the form (# of rows with "TotalTournaments" = 0, # of columns)
print(df[df["TotalTournaments"] != 0].shape) # Prints (# of rows with "TotalTournaments" != 0, # of columns)
df = df[df["TotalTournaments"] != 0]

(48, 8)
(621, 8)


Start with removing duplicate entries, to avoid skewed analytical data. The following two `print` statements are similar to the previous one: the first shows the number of entries that have a value of 0 under "TotalTournaments", which is 48, and the second shows entries that have a *non-zero* value under "TotalTournaments", which is 621. Since our analysis is only concerned with games that have had eSport activity (meaning tournaments were held), we can remove the entries with values of 0 for "TotalTournaments"; this was done in the last line. We can verify that the able no longer has these values by the following code:

In [4]:
print((df["TotalTournaments"] == 0).any()) # Prints a check if any entries with "TotalTournaments" == 0 exist; True if at least one exists, False if none exist

False


To allow for more compact visualization, we will shorten the genre names by their respective abbreviations, or removing some unneseccary words.
<br>
- "First-Person Shooter" to "FPS"
- "Multiplayer Online Battle Arena" to "MOBA"
- "Role-Playing Game" to "RPG"
- "Fighting Game" to "Fighting"
- "Collectible Card Game" to "CCG"
- "Puzzle Game" to "Puzzle"
- "Battle Royale" to "BR"
- "Third-Person Shooter" to "TPS"
- "Music / Rhythm Game" to "Rhythm"

"Strategy", "Racing", and "Sports" do not need to be changed.

In [ ]:
df["Genre"] = df["Genre"].replace({ # replaces each "Genre" value with the following shortened value, all at once
    "First-Person Shooter": "FPS",
    "Multiplayer Online Battle Arena": "MOBA",
    "Role-Playing Game": "RPG",
    "Fighting Game": "Fighting",
    "Collectible Card Game": "CCG",
    "Puzzle Game": "Puzzle",
    "Battle Royale": "BR",
    "Third-Person Shooter": "TPS",
    "Music / Rhythm Game": "Rhythm"
})
df["Genre"].unique() # Print all of the unique values under the "Genre" column to verify they have changed

array(['Strategy', 'FPS', 'MOBA', 'RPG', 'Fighting', 'Racing', 'Sports',
       'CCG', 'Puzzle', 'BR', 'TPS', 'Rhythm'], dtype=object)